unstructured_api_tools/pipelines/templates/pipeline_api.txt

```diff
@@ -221,7 +226,18 @@ async def pipeline_1(request: Request,
             def response_generator():
                 for file in {{var_name}}:
                     {% if accepts_text %}
-                    text = file.file.read().decode("utf-8")
+                    {% if default_gz_uncompressed_content_type %} 
```

Comment:
A few things:

* Per prior comments, there is no `default_gz_uncompressed_content_type`, so all newly added logic should be in plain old python rather than jinja template.
* in the `accepts_text` case, to determine if it is compressed you can first check `file.content_type` (see https://fastapi.tiangolo.com/tutorial/request-files/#uploadfile ) for the mimetype first to see if it is `application/gzip`, then do the magic thing. if not `application/gzip`, no change is needed from existing code. otherwise decompress it.
* the logic is similar for non-text `file` case, only if the uploaded file is `application/gzip`, use `gz_uncompressed_content_type` if not `None`, otherwise the magic thing.